docs/how-to/dummy-data.md

```diff
@@ -0,0 +1,144 @@
+
+There are 3 ways to use dummy data in ehrQL.
+
+1. Let ehrQL [generate your dummy dataset from your dataset definition](#generate-a-dummy-dataset-from-a-dataset-definition)
+
+1. Supply a [dummy dataset file](#supply-your-own-dummy-dataset-file)
+
+1. Supply a [set of dummy tables](#supply-your-own-dummy-tables)
+
+
+## Generate a dummy dataset from a dataset definition
+
+ehrQL is designed so that the same command can be used to output a dummy dataset when run on your own computer and then output a real dataset when run inside the secure environment as part of an OpenSAFELY pipeline.
+
+Refer to the [documentation on generating a dummy dataset](../../tutorial/generating-a-dummy-dataset/#generate-a-dummy-dataset) to see how this works.
+
+You do not need to add anything to the dataset definition itself in order to generate a dummy
+dataset in this way. ehrQL will use the dataset definition to set up dummy data and generate
+matching patients.
+
+By default, 500 patients will be generated in a dummy dataset. If you need to increase this
+number, you can configure it in the dataset definition with:
+
+```
+dataset.configure_dummy_data(population_size=1000)
+```
+
+:warning: Increasing the population size will increase the time required to generate the
+dataset.
+
+
+## Supply your own dummy dataset file
+
+You can provide a dummy dataset file in the following formats.
+
+|Format        |File extension|
+|--------------|--------------|
+|CSV           |.csv          |
+|Compressed CSV|.csv.gz       |
+|Arrow         |.arrow        |
+
+:warning: Your dummy dataset file must have the relevant file extension shown in the table
+above.
+
+For example, take this dataset definition from the tutorial:
+
+```ehrql
+from ehrql import create_dataset
+from ehrql.tables.beta.core import patients, medications
+
+dataset = create_dataset()
+
+dataset.define_population(patients.date_of_birth.is_on_or_before("1999-12-31"))
+
+asthma_codes = ["39113311000001107", "39113611000001102"]
+latest_asthma_med = (
+    medications.where(medications.dmd_code.is_in(asthma_codes))
+    .sort_by(medications.date)
+    .last_for_patient()
+)
+
+dataset.asthma_med_date = latest_asthma_med.date
+dataset.asthma_med_code = latest_asthma_med.dmd_code
+```
+
+And this dummy dataset, in a CSV file named `dummy.csv`:
+
+|patient_id|asthma_med_date|asthma_med_code  |
+|----------|---------------|-----------------|
+|1         |2021-01-01     |39113311000001107|
+|2         |2020-03-10     |39113311000001107|
+|3         |2022-11-22     |39113311000001107|
+|4         |2021-06-07     |39113611000001102|
+|5         |2023-10-17     |39113611000001102|
+
+Run the dataset definition with the dummy dataset file:
+
+```
+opensafely exec ehrql:v0 generate-dataset dataset_definition.py --dummy-data-file dummy.csv
+```
+
+Now, instead of a generated dummy dataset, you'll see the data from the dummy data file that
+you provided.
```

Comment:
I think we should mention that ehrQL will check the column names, types, and categorical values.